applications/StructuralMechanicsApplication/tests/cpp_tests/test_solid_transient_sensitivity.cpp

```diff
@@ -0,0 +1,276 @@
+// KRATOS  ___|  |                   |                   |
+//       \___ \  __|  __| |   |  __| __| |   |  __| _` | |
+//             | |   |    |   | (    |   |   | |   (   | |
+//       _____/ \__|_|   \__,_|\___|\__|\__,_|_|  \__,_|_| MECHANICS
+//
+//  License:		 BSD License
+//					 license: structural_mechanics_application/license.txt
+//
+//  Main authors:
+//
+
+// System includes
+#include <functional>
+#include <stack>
+#include <string>
+#include <unordered_map>
+
+// External includes
+
+// Project includes
+#include "containers/model.h"
+#include "includes/shared_pointers.h"
+#include "linear_solvers/skyline_lu_custom_scalar_solver.h"
+#include "solving_strategies/convergencecriterias/residual_criteria.h"
+#include "solving_strategies/schemes/residual_based_adjoint_bossak_scheme.h"
+#include "solving_strategies/schemes/residual_based_bossak_displacement_scheme.hpp"
+#include "solving_strategies/strategies/residualbased_linear_strategy.h"
+#include "solving_strategies/strategies/residualbased_newton_raphson_strategy.h"
+#include "spaces/ublas_space.h"
+#include "testing/testing.h"
+#include "utilities/sensitivity_builder.h"
+
+// Application includes
+#include "custom_response_functions/response_utilities/adjoint_nodal_displacement_response_function.h"
+#include "tests/cpp_tests/bits/adjoint_test_model_part_factory.h"
+#include "tests/cpp_tests/bits/test_model_part_factory.h"
+
+namespace Kratos
+{
+namespace
+{
+namespace smtsts
+{ // cotire unity guard
+struct PrimalTestSolver
+{
+    std::function<ModelPart&()> ModelPartFactory;
+    unsigned ResponseNodeId;
+    double TotalTime;
+    double TimeStepSize;
+
+    double CalculateResponseValue();
+
+    double CalculateResponseValue(unsigned NodeToPerturb, char Direction, double Perturbation);
+};
+
+struct AdjointTestSolver
+{
+    std::function<ModelPart&()> ModelPartFactory;
+    unsigned ResponseNodeId;
+    double TotalTime;
+    double TimeStepSize;
+
+    double CalculateSensitivity(unsigned NodeToPerturb, char Direction);
+};
+
+} // namespace smtsts
+} // namespace
+
+namespace Testing
+{
+KRATOS_TEST_CASE_IN_SUITE(TotalLagrangian2D3_TransientSensitivity, KratosStructuralMechanicsFastSuite)
+{
+    Model this_model;
+    auto model_part_factory = [&this_model]() -> ModelPart& {
+        return CreateStructuralMechanicsTestModelPart(
+            &this_model, KratosComponents<Element>::Get("TotalLagrangianElement2D3N"),
+            KratosComponents<ConstitutiveLaw>::Get(
+                "LinearElasticPlaneStrain2DLaw"),
+            [](ModelPart* pModelPart) {
+                pModelPart->GetNode(1).Fix(DISPLACEMENT_X);
+                pModelPart->GetNode(1).Fix(DISPLACEMENT_Y);
+                pModelPart->GetNode(3).Fix(DISPLACEMENT_X);
+                pModelPart->GetNode(3).Fix(DISPLACEMENT_Y);
+            });
+    };
+    const double time_step_size = 0.004;
+    const double total_time = 0.015; // approx. 1/4 of a period.
+    const unsigned response_node_id = 2;
```

Comment:
maybe i am missing something, but is there anything happening during the time loop? Where are loads applied?

Reply:
The element has a body force assigned in the factory.